src/chapter-03/src/main.rs

```diff
@@ -20,48 +19,42 @@ async fn function_handler(
             let shorten_url_request_body = event.payload::<ShortenUrlRequest>()?;
 
             match shorten_url_request_body {
-                None => {
-                    let resp = generate_api_response(400, "".to_string());
-
-                    resp
-                }
+                None => generate_api_response(400, "Bad Request"),
                 Some(shorten_url_request) => {
                     let shortened_url_response = url_shortener.shorten_url(shorten_url_request);
-
-                    let response = match shortened_url_response {
-                        Ok(response) => {
-                            generate_api_response(200, serde_json::to_string(&response).unwrap())?
-                        }
-                        Err(_) => generate_api_response(400, "".to_string())?,
-                    };
-
-                    Ok(response)
+                    Ok(generate_api_response(
```

Comment:
This simplification derives by the fact that we have removed the `Result` in the return type of `shorten_url()`.